File: domain/usecases/get_group_packages.py
from typing import List
from domain.entities import Package
from domain.repositories import PackageRepository

class GetGroupPackagesUseCase:
    def __init__(self, alpm_repo: PackageRepository):
        self.alpm_repo = alpm_repo

    def execute(self, group_name: str) -> List[Package]:
        if not group_name:
            return []
        try:
            return self.alpm_repo.get_packages_by_group(group_name)
        except Exception as e:
            print(f"Error getting packages for group {group_name}: {e}")
            return []

File: src-rust/src/domain/usecases/get_group_packages.rs
use std::sync::Arc;
use crate::domain::entities::Package;

pub struct GetGroupPackagesUseCase {
    pub alpm_repo: Arc<crate::data::alpm_repository::AlpmPackageRepository>,
}

impl GetGroupPackagesUseCase {
    pub fn new(alpm_repo: Arc<crate::data::alpm_repository::AlpmPackageRepository>) -> Self {
        Self { alpm_repo }
    }

    pub async fn execute(&self, group_name: &str) -> Vec<Package> {
        self.alpm_repo.get_packages_by_group(group_name).await
    }
}
